### DISCUSSION TITLE
How are plans to be used?

### DISCUSSION THREAD

**Discussion by evandromurilo** [plandex-ai/plandex#60] [General]
Hi there!

I'm using plandex daily but still don't fully understand the concept of a plan. Do I want to use a single plan for the lifetime of a project? I did that the first time, and at the end of the day it seemed to have degraded and was not so effective anymore, so I created a new plan and it was smart again (maybe I messed up with a few rewinds). Or should I use one plan for each series of related tasks (one for issue)?

As I'm telling it to do things, does it remembers the previous tasks? Or is the context clean for each task, meaning it uses only the files added + the task text?

**Comment by danenania** [comment]
Hi @evandromurilo! You can maybe think of a plan as being similar to a single conversation in ChatGPT. 

Plandex does keep the conversation in context up  to the `Max Convo Tokens` setting (which you can see with`plandex models` or change with `plandex set-model`), and beyond that it gradually summarizes the earliest messages to stay under the limit.

In general, I tend to use separate plans for separate tasks unless there is some shared context between them or a task follows on directly from another one. It’s better to keep context smaller if you can, both for model performance and reducing costs.
